### PR TITLE
Add MAAS infrastructure tab for CAPI tutorial

### PR DESCRIPTION
**Pull request by berkayoz**
This PR adds a tab for the MAAS infrastructure in the CAPI tutorial.

### CODE REVIEW

**Review by bschimke95 — COMMENTED** [review]
LGTM, leave approval to @evilnick

File: docs/src/capi/tutorial/getting-started.md [hunk 1]
```diff
@@ -94,7 +94,49 @@ export AWS_B64ENCODED_CREDENTIALS=$(clusterawsadm bootstrap credentials encode-a
...
+````
+
+````{group-tab} MAAS
+Start by setting up environment variables defining the MAAS credentials, if
+these are not already defined:
+
+```
+export MAAS_API_KEY="<maas-api-key>"
+export MAAS_ENDPOINT="http://<maas-endpoint>/MAAS"
+export MAAS_DNS_DOMAIN="<maas-dns-domain>"
```
Comment (addyess) [inline docs/src/capi/tutorial/getting-started.md:106]: Is there anything more necessary to do if maas_endpoint is `https`?  Is that supported?

File: docs/src/capi/tutorial/getting-started.md [hunk 2]
```diff
@@ -94,7 +94,49 @@ export AWS_B64ENCODED_CREDENTIALS=$(clusterawsadm bootstrap credentials encode-a
 ```
 
 You are now all set to deploy the AWS CAPI infrastructure provider.
```
Comment (eaudetcobello) [inline docs/src/capi/tutorial/getting-started.md:96]:
```suggestion
You are now all set to deploy the MAAS infrastructure provider.
```

**Review by eaudetcobello — APPROVED** [review]

**Review by eaudetcobello — CHANGES_REQUESTED** [review]
revoke approval

File: docs/src/capi/tutorial/getting-started.md (hunk 2 above)
Comment (bschimke95) [reply docs/src/capi/tutorial/getting-started.md:96]: That is the tab of the AWS infrastructure provider. So, this should be correct. 
The MAAS one is below.

File: docs/src/capi/tutorial/getting-started.md [hunk 3]
```diff
@@ -94,7 +94,49 @@ export AWS_B64ENCODED_CREDENTIALS=$(clusterawsadm bootstrap credentials encode-a
...
+````
+
+````{group-tab} MAAS
+Start by setting up environment variables defining the MAAS credentials, if
+these are not already defined:
```
Comment (louiseschmidtgen) [inline docs/src/capi/tutorial/getting-started.md:101]:
```suggestion
Start by setting up environment variables to allow access to MAAS:
```

File: docs/src/capi/tutorial/getting-started.md [hunk 4]
```diff
@@ -94,7 +94,49 @@ export AWS_B64ENCODED_CREDENTIALS=$(clusterawsadm bootstrap credentials encode-a
...
+export MAAS_DNS_DOMAIN="<maas-dns-domain>"
+```
+The MAAS infrastructure provider uses these credentials to deploy machines, create DNS records and perform various other operations for workload clusters.
+
+```{warning}
+The management cluster needs to be able to resolve dns records from the MAAS domain.
+This usually means the management cluster also needs to be deployed on a MAAS machine.
+```
+
+Continue with setting up environment variables defining the machine image and minimum compute resources of the control plane and worker nodes:
```
Comment (louiseschmidtgen) [inline docs/src/capi/tutorial/getting-started.md:115]:
```suggestion
Define further environment variables for the machine image and minimum compute resources of the control plane and worker nodes:
```

File: docs/src/capi/tutorial/getting-started.md [hunk 5]
```diff
@@ -94,7 +94,49 @@ export AWS_B64ENCODED_CREDENTIALS=$(clusterawsadm bootstrap credentials encode-a
...
+export CONTROL_PLANE_MACHINE_MINCPU="2"
+export CONTROL_PLANE_MACHINE_MINMEMORY="4096"
+export CONTROL_PLANE_MACHINE_IMAGE="ubuntu"
+
+export WORKER_MACHINE_MINCPU="2"
+export WORKER_MACHINE_MINMEMORY="4096"
+export WORKER_MACHINE_IMAGE="ubuntu"
+```
+
+Optional environment variables can be defined for specifying resource pools and machine tags like:
```
Comment (louiseschmidtgen) [inline docs/src/capi/tutorial/getting-started.md:127]:
```suggestion
Optional environment variables can be defined for specifying resource pools and machine tags:
```

**Review by louiseschmidtgen — COMMENTED** [review]
Nice work @berkayoz , I've added a few suggestions to improve the flow of words.

File: docs/src/capi/tutorial/getting-started.md [hunk 6]
```diff
@@ -94,7 +94,49 @@ export AWS_B64ENCODED_CREDENTIALS=$(clusterawsadm bootstrap credentials encode-a
...
+
+Continue with setting up environment variables defining the machine image and minimum compute resources of the control plane and worker nodes:
+
+```
+export CONTROL_PLANE_MACHINE_MINCPU="2"
+export CONTROL_PLANE_MACHINE_MINMEMORY="4096"
+export CONTROL_PLANE_MACHINE_IMAGE="ubuntu"
+
+export WORKER_MACHINE_MINCPU="2"
+export WORKER_MACHINE_MINMEMORY="4096"
```
Comment (eaudetcobello) [inline docs/src/capi/tutorial/getting-started.md:123]: I think 4G of memory is a bit high. I'd set it to 2048 by default and the user can choose if they want more.

**Review by eaudetcobello — CHANGES_REQUESTED** [review]

File: docs/src/capi/tutorial/getting-started.md (hunk 1 above)
Comment (berkayoz) [reply docs/src/capi/tutorial/getting-started.md:106]: AFAIK upstream provider does not suggest/do anything special for `https` endpoints. The provider uses a simple go client which performs http requests through the standard library so I expect things to work properly. Seems like the client uses `InsecureSkipVerify: true` which can be seen [here](https://github.com/spectrocloud/maas-client-go/blob/072142d8aa50f39a4f11cd59e3a83a53f7f047c4/maasclient/client.go#L244).

**Review by evilnick — APPROVED** [review]
made some fixes. Thanks!

**Review by louiseschmidtgen — APPROVED** [review]
LGTM

**Review by eaudetcobello — APPROVED** [review]

**Pull request merged**